tgui/packages/tgui/interfaces/BigManipulator.tsx

```diff
@@ -1,29 +1,123 @@
-  const { active } = data;
+  const {
+    active,
+    manipulate_mode,
+    settings_list,
+    drop_after_use,
+    highest_priority,
+    throw_range,
+    item_as_filter,
+    selected_type,
+  } = data;
   return (
-    <Window title="Manipulator Interface" width={320} height={160}>
+    <Window title="Manipulator Interface" width={320} height={240}>
       <Window.Content>
-        <Section title="Action panel">
-          <Stack>
+        <Section
+          title="Action panel"
+          buttons={
             <Button
               icon="power-off"
               content={active ? 'On' : 'Off'}
               selected={active}
               onClick={() => act('on')}
             />
-          </Stack>
+          }
+        >
+          <Stack.Item grow>
+            <Button
+              content={`type to take: ${selected_type}`}
+              tooltip="changes type of item that manipulator will pick up."
+              onClick={() => act('change_take_item_type')}
+            />
+            <Button
+              content={`Mode: ${manipulate_mode}`}
+              tooltip="click to change manipulator mode"
+              onClick={() => act('change_mode')}
+            />
+            <Button
+              icon="eject"
+              tooltip="drop item contained in manipulator"
+              onClick={() => act('drop')}
+            />
+          </Stack.Item>
+          <Stack.Item grow>
+            <Button
+              content={`filter: ${item_as_filter ? `${item_as_filter}` : `no filter`}`}
+              tooltip="click on this button with item in hands to add filter on this item."
+              onClick={() => act('add_filter')}
+            />
+            {manipulate_mode !== 'Use' ? undefined : (
+              <Button
+                content={'Drop Use'}
+                tooltip="drop item after use. othewise manipulator will use this item after cooldown."
+                selected={drop_after_use}
+                onClick={() => act('drop_use_change')}
+              />
+            )}
+            {manipulate_mode !== 'Throw' ? undefined : (
+              <Button
+                content={`Throw range: ${throw_range}`}
+                tooltip="distance an object will travel when thrown"
+                onClick={() => act('change_throw_range')}
+              />
+            )}
+          </Stack.Item>
         </Section>
+        {settings_list.length < 2 ? undefined : (
+          <Section fill scrollable>
+            {settings_list.length < 2 ? undefined : (
+              <Button
+                content={'Only 1 priority'}
+                selected={highest_priority}
+                tooltip="manipulate only on 1 priority in priority list"
+                onClick={() => act('highest_priority_change')}
+              />
+            )}
+            <Table>
+              {settings_list.map((setting) => (
+                <Table.Row key={setting.priority_width} className="candystripe">
```

Comment:
are you sure `priority_width` will be unique?

Reply:
Must be unique because otherwise something might break. I'll change it to name just to be sure.